Jenkinsfile

```diff
@@ -13,177 +13,158 @@
-              sh "curl --fail -u '${CHARTMUSEUM_AUTH}' --data-binary '@${CHART_NAME}-${nextVersion}.tgz' ${CHART_REPOSITORY}/api/charts"
+            def archive = "${CHART_NAME}-${BUILD_VERSION}.tgz"
+            echo "Upload chart archive ${archive}"
+            // upload package to the ChartMuseum
+            withCredentials([usernameColonPassword(credentialsId: 'jenkins-x-chartmuseum', variable: 'CHARTMUSEUM_AUTH')]) {
+              sh "curl --fail -u '${CHARTMUSEUM_AUTH}' --data-binary '@${archive}' ${CHART_REPOSITORY}/api/charts"
             }
           }
         }
       }
       post {
         always {
           container('jx-base') {
-            // uninstall the chart
-            sh """
-              jx step helm delete ${TEST_RELEASE} \
-                --namespace=${TEST_NAMESPACE} \
-                --purge
-            """
-            // clean up the test namespace
-            sh "kubectl delete namespace ${TEST_NAMESPACE} --ignore-not-found=true"
+            script {
+              try {
+                // uninstall the chart
+                sh "helm3 uninstall ${TEST_RELEASE} --namespace=${TEST_NAMESPACE}"
+              } finally {
+                // clean up the test namespace
+                sh "kubectl delete namespace ${TEST_NAMESPACE} --ignore-not-found=true"
+              }
+            }
           }
         }
         success {
-          setGitHubBuildStatus('helm-release', 'Build and release Helm chart', 'SUCCESS')
+          setGitHubBuildStatus('package', 'Package and upload Helm chart', 'SUCCESS')
         }
         failure {
-          setGitHubBuildStatus('helm-release', 'Build and release Helm chart', 'FAILURE')
+          setGitHubBuildStatus('package', 'Package and upload Helm chart', 'FAILURE')
         }
       }
     }
     stage('GitHub release') {
       when {
-        branch 'master'
+        allOf {
+          branch 'master'
+          not {
+            environment name: 'DRY_RUN', value: 'true'
+          }
+        }
       }
       steps {
+        setGitHubBuildStatus('release', 'Release', 'PENDING')
         container('jx-base') {
           script {
-            def currentNamespace = sh(returnStdout: true, script: "jx --batch-mode ns | cut -d\\' -f2").trim()
-            if (currentNamespace == 'platform-staging') {
-              echo "Running in namespace ${currentNamespace}, skip GitHub release stage."
-              return
-            }
-            setGitHubBuildStatus('github-release', 'GitHub release', 'PENDING')
             sh """
+              # add release commit
+              git commit -a -m "Release ${BUILD_VERSION}"
+
               # create the Git credentials
               jx step git credentials
               git config credential.helper store
 
-              # push release commit added by the revious stage to master
+              # push release commit to master
               git push origin master:master
```

Comment:
I saw that you removed the aborted stage, won't we have an infinite build/release on master due to this push?

Reply:
Yes indeed, thanks. Switched to a SNAPSHOT version and no more commit to the master branch, only pushing a tag pointing to a commit with the bumped version.